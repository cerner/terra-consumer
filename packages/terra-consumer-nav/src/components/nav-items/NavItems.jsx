import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import NavItem from './NavItem';
import styles from './NavItem.scss';

const cx = classNames.bind(styles);

const propTypes = {
  navItems: PropTypes.arrayOf(PropTypes.object),
};

const defaultProps = {
  navItems: [],
};

class NavItems extends Component {
  constructor() {
    super();

    this.state = {
      openToggle: null,
    };

    this.handleToggle = this.handleToggle.bind(this);
  }

  handleToggle(i, isOpen) {
    this.setState({
      openToggle: isOpen ? i : null,
    });
  }

  render() {
    const { navItems, ...customProps } = this.props;

    const content = navItems.map((element, i) => {
      let toggleProps = {};
      let subNavs = [];
      if (element.subItems) {
        subNavs = element.subItems.map(item => <NavItem key={item.text} {...item} />);
        toggleProps = {
          isOpen: this.state.openToggle === i,
          handleToggle: this.handleToggle,
          toggleId: i,
        };
      }

      return (
        <NavItem key={element.text} {...element} {...toggleProps}>
          {subNavs}
        </NavItem>
      );
    });

    return (
      <div className={cx('nav-items-contatiner')} {...customProps}>
        {content}
      </div>
    );
  }
}

NavItems.propTypes = propTypes;
NavItems.defaultProps = defaultProps;

export default NavItems;
