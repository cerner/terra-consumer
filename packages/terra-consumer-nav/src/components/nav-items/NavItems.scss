:local {
  .nav-icon {
    height: 1rem;
    width: 1rem;
  }

  .nav-item {
    color: #3d3d3d;
    font-size: 16px;
    text-decoration: none;
  }
}
