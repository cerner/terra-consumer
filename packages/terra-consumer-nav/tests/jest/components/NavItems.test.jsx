import React from 'react';
import NavItems from '../../../src/components/nav-items/NavItems';

const testData = {
  navItems: [{
    uri: '#testPath',
    text: 'testIcon',
    isActive: false,
    badgeValue: 0,
  }],
};

describe('Nav Items', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(<NavItems {...testData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should apply custom classes', () => {
    const wrapper = shallow(<NavItems {...testData} className="test-class" />);
    expect(wrapper).toMatchSnapshot();
  });
});
