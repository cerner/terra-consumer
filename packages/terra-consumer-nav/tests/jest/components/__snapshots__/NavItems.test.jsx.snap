// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Items should apply custom classes 1`] = `
<div
  className="test-class"
>
  <NavItem
    badgeValue={0}
    handleToggle={[Function]}
    isActive={false}
    text="testIcon"
    url="#testPath"
  />
</div>
`;

exports[`Nav Items should render a default component 1`] = `
<div
  className=""
>
  <NavItem
    badgeValue={0}
    handleToggle={[Function]}
    isActive={false}
    text="testIcon"
    url="#testPath"
  />
</div>
`;
